source/module_base/complexarray.h

```diff
@@ -83,23 +89,37 @@ class ComplexArray
 ComplexArray operator*(double r, const ComplexArray &cd);
 ComplexArray operator*(std::complex < double> c, const ComplexArray &cd);
 
+//Sum of absolute squares of all elements in cd
 double abs2(const ComplexArray &cd);
 
+
 void
 add_scale_abs2(const std::complex < double> &c, const ComplexArray & in,
                ComplexArray &out);
 
+// Take "dot-product" of two ComplexArray:  sum of cd1(conjugate)[i] * cd2[i]
```

Comment:
I'm sorry but why conjugate on cd1 in dot product?

Reply:
i also don't know why. this is just what the code represented.

Reply:
> i also don't know why. this is just what the code represented.

That's weird, did this function used in code somewhere else?

Reply:
I can understand this action, which is really common in quantum chemistry like (psi^* * psi), of cause this is not a standard dot-product, so the comment is "dot-product". I do not know how to name this action, so I just copy it from .cpp file, and add my my understanding in the last half sentence .

Reply:
> I can understand this action, which is really common in quantum chemistry like (psi^* * psi), of cause this is not a standard dot-product, so the comment is "dot-product". I do not know how to name this action, so I just copy it from .cpp file, and add my my understanding in the last half sentence .

OK, do we need a new name here, or leave it as is? I suggest changing a more unique name avoiding confusing.

Reply:
My previous understanding is wrong, after checking with wenfei-li,  this action is the standard dot-product of two complex array.

Reply:
> My previous understanding is wrong, after checking with wenfei-li, this action is the standard dot-product of two complex array.

You are right.
Ref: https://www.mathworks.com/help/matlab/ref/dot.html
> When inputs A and B are real or complex vectors, the dot function treats them as column vectors and dot(A,B) is the same as sum(conj(A).*B).